plugins/experimental/uri_signing/uri_signing.c

```diff
@@ -218,6 +224,32 @@ TSRemapDoRemap(void *ih, TSHttpTxn txnp, TSRemapRequestInfo *rri)
       checkpoints[cpi++] = mark_timer(&t);
     }
     jws = get_jws_from_cookie(&client_cookie, &client_cookie_sz_ct, package);
+  } else {
+    /* There has been a JWS found in the url */
+    /* Strip the token from the URL for upstream if configured to do so */
+    if (config_strip_token((struct config *)ih)) {
+      if (strncmp(url, strip_uri, strip_ct) != 0) {
+        char *strip_uri_start = &strip_uri[0];
+        char *strip_uri_end   = &strip_uri[strip_ct];
+        PluginDebug("Stripping token from upstream url to: %s", strip_uri_start);
+        if (TSUrlParse(rri->requestBufp, rri->requestUrl, (const char **)&strip_uri_start, strip_uri_end) != TS_PARSE_DONE) {
+          PluginDebug("Error in TSUrlParse");
+        } else {
+          status = TSREMAP_DID_REMAP;
+        }
+      }
+    }
+
+    /* Check auth_dir and pass through if configured */
+    if (uri_matches_auth_directive((struct config *)ih, url, url_ct)) {
+      if (url != NULL) {
+        TSfree((void *)url);
+      }
+      if (url != NULL) {
+        free(strip_uri);
+      }
+      return TSREMAP_NO_REMAP;
```

Comment:
Should this return `status`, in case a token was stripped?

Reply:
If I modify the URL using TSUrlParse and return TSREMAP_NO_REMAP I am seeing the expected behavior of the token being stripped and the stripped URL reaching the origin. This doesn't seem to be the correct usage though. 

If I modify the URL using TSUrlParse and then return a TSREMAP_DID_REMAP as I believe I should, I am getting 405s and nothing hits the origin. I am trying to track down what is going on.

Reply:
I believe the plugin is setting the working URL to the stripped pristine URL and then returning a TSREMAP_DID_REMAP is preventing the core from remapping to whatever is in remap.config. We are then trying to send the pristine URL upstream and failing. 

Leaving as TSREMAP_NO_REMAP will work for what I've tested thus far, but who knows how that will behave with other configurations/plugins. 

I think you're right and we should be parsing and stripping from the remapped URL rather than pristine (however we can get that). 

https://github.com/apache/trafficserver/issues/4026 
The above issue might complicate things if uri_signing is to be the first plugin.

Reply:
A discussion is currently being had on the dev mailing list regarding if the core should remap before or after the plugins. If the core runs before the plugins, I would be able to assume that the requestUrl has been remapped, and return TS_DID_REMAP here with the code as is. 

If the plugins run before the core, it would be very difficult to get hands on the remapped request with query strings and paths in a remap plugin. I think I would have to piece together the URL I'd like to set from a mix of remap config values and the path/query strings from the existing url. I would then be able to return TS_DID_REMAP. This second way seems cumbersome and like something could easily be forgotten or left out when piecing together the remapped URL. Additionally, we'd be moving integral processing usually handled by the core into the plugin. 

I am going to wait a bit to see how that pans out.